EEJT-0003/src/main/java/com/epam/esm/giftcertficate/service/GiftCertificateService.java

```diff
@@ -0,0 +1,220 @@
+        log.info("Service receives params for getting");
+        List<GiftCertificate> giftCertificatesByTagsAndPrice = giftCertificateRepository.
+                getGiftCertificatesByTagsAndPrice(firstTagName, secondTagName, price,
+                        getPaginationBegin(page, size),
+                        getPaginationEnd(page, size));
+
+        log.debug("Service returns gift-certificates");
+        return new PageImpl<>(giftCertificatesByTagsAndPrice);
+    }
+
+    public Page<GiftCertificate> getGiftCertificatesByNameOrByPartOfName(String partOfName, int page, int size) {
+        log.info("Service receives params for getting");
+
+        List<GiftCertificate> allGiftCertificateByPartOfName = giftCertificateRepository.
+                getAllGiftCertificateByPartOfName(partOfName + PERCENT_FOR_SEARCHING_BY_PART,
+                        getPaginationBegin(page, size),
+                        getPaginationEnd(page, size));
+
+        log.debug("Service returns gift-certificates");
+        return new PageImpl<>(allGiftCertificateByPartOfName);
+    }
+
+    public Page<GiftCertificate> getGiftCertificatesSortedByDate(String sortDirection, int page, int size) {
+        log.info("Service receives params for getting");
+
+        PageRequest pageRequest = PageRequest.of(page, size, Sort.by(Sort.Direction.valueOf(sortDirection.toUpperCase(Locale.ROOT)), CREATE_DATE));
+
+        log.debug("Service returns gift-certificates");
+        return giftCertificateRepository.findAll(pageRequest);
+    }
+
+    public Page<GiftCertificate> getGiftCertificatesSortedByDateAndByName(String firstSortDirection, String secondSortDirection, int page, int size) {
+        log.info("Service receives params for getting");
+
+        PageRequest pageRequest = PageRequest.of(page, size, Sort.by(Sort.Direction.valueOf(firstSortDirection.toUpperCase(Locale.ROOT)), NAME)
+                .and(Sort.by(Sort.Direction.valueOf(secondSortDirection.toUpperCase(Locale.ROOT)), CREATE_DATE)));
+
+        log.debug("Service returns gift-certificates");
+        return giftCertificateRepository.findAll(pageRequest);
+    }
+
+    private GiftCertificate updatingGiftCertificateUsingMapOfUpdates(Map<String, String> updatesMap, GiftCertificate gifCertificateForUpdating) {
+        updatesMap.forEach((key, value) -> {
+            if (key.equals("name")) {
+                gifCertificateForUpdating.setName(value);
+            } else if (key.equals("description")) {
+                gifCertificateForUpdating.setDescription(value);
+            } else if (key.equals("price")) {
+                gifCertificateForUpdating.setPrice(Integer.parseInt(value));
+            } else if (key.equals("duration")) {
+                gifCertificateForUpdating.setDurationDays(Integer.parseInt(value));
+            }
+        });
+        return gifCertificateForUpdating;
+    }
```

Comment:
I would suggest to rewrite this without cycle  
You can try to use this method to increase readability:  
```java
    private <T>void setIfNotNull(T value, Consumer<T> setter) {
        if (value != null) {
            setter.accept(value);
        }
    }
```
With it you can write
```java
        setIfNotNull(updatesMap.get("name"), gifCertificateForUpdating::setName);
        setIfNotNull(updatesMap.get("description"), gifCertificateForUpdating::setDescription);
        //...
```

Also I would suggest to not use updatesMap, but GiftCertificate object instead